### PR TITLE
Add makeRequests and processRequest for groups

### PR DESCRIPTION
**Pull request by jarrydlee**
fixes https://github.com/PagerNation/PagerNation/issues/104

### CODE REVIEW

File: server/models/group.js [hunk 1]
```diff
@@ -21,6 +21,11 @@ const GroupSchema = new mongoose.Schema({
     type: EscalationPolicy.schema,
     default: EscalationPolicy.defaultEscalationPolicy()
   },
+  requests: [{
```
Comment (kalebdavis) [inline server/models/group.js]: Can you make this something a bit more verbose? Something like `userRequests` or `joinRequests`? Request could mean way too many things.

File: server/services/group.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ import JoiHelper from '../helpers/JoiHelper';
 import Group from '../models/group';
 import userService from './user';
 
+const ID_SCHEMA = Joi.string().hex().length(24);
```
Comment (kalebdavis) [inline server/services/group.js]: Nice

File: server/services/group.js [hunk 3]
```diff
@@ -62,11 +61,27 @@ function removeUser(group, userId) {
...
+    .then(id => Group.makeRequest(groupName, id));
+}
+
+function processRequest(group, userId, isAcceted) {
+  const schema = Joi.object().keys({
+    userId: ID_SCHEMA.required(),
+    isAcceted: Joi.boolean().required()
+  });
+
+  return JoiHelper.validate({ userId, isAcceted }, schema)
```
Comment (kalebdavis) [inline server/services/group.js]: nit: Spelling `isAcceted`

File: server/tests/routes/group.test.js [hunk 4]
```diff
@@ -329,4 +329,95 @@ describe('## Group API', () => {
...
+          admins: [values[0].user.id]
+        })))
+        .then((g) => {
+          group = g;
+          done();
+        });
+    });
+
+    context('with group admin permissions', () => {
+      it('accepts a request', (done) => {
```
Comment (kalebdavis) [inline server/tests/routes/group.test.js]: Can you add a test for rejecting the request through the route?

File: server/tests/routes/group.test.js [hunk 5]
```diff
@@ -329,4 +329,95 @@ describe('## Group API', () => {
...
+            expect(res.body.requests).to.be.empty;
+            expect(res.body.users).to.include(userAndToken.user.id);
+            done();
+          });
+      });
+    });
+
+    context('with admin permissions', () => {
+      it('accepts a request', (done) => {
+        request(app);
```
Comment (kalebdavis) [inline server/tests/routes/group.test.js]: I don't think this line should be here

File: server/tests/routes/group.test.js [hunk 6]
```diff
@@ -329,4 +329,95 @@ describe('## Group API', () => {
...
+          .send({
+            userId: userAndToken.user.id,
+            isAccepted: true
+          })
+          .expect(httpStatus.UNAUTHORIZED)
+          .then(() => done());
+      });
+    });
+
+    context('without a nonexistent user', () => {
```
Comment (kalebdavis) [inline server/tests/routes/group.test.js]: nit: 'with a nonexistent user'

File: server/tests/routes/group.test.js (hunk 4 above)
Comment (jarrydlee) [inline server/tests/routes/group.test.js]: The functionality between the service and the route isn't different, the route tests were made just to test the permissions on the route.

**Pull request merged**